rcsbsearchapi/search.py

```diff
@@ -188,91 +189,72 @@ def or_(self, other: Union[str, "Query", "Attr"], qtype=STRUCTURE_ATTRIBUTE_SEAR
 
 @dataclass(frozen=True)
 class Terminal(Query):
-    """A terminal query node.
-
-    Terminals are simple predicates comparing some *attribute* of a structure to a
-    value.
-
-    Examples:
-        >>> Terminal("exptl.method", "exact_match", "X-RAY DIFFRACTION")
-        >>> Terminal("rcsb_id", "in", ["5T89", "1TIM"])
-        >>> Terminal(value="tubulin")
-
-    A full list of attributes is available in the
-    `schema <http://search.rcsb.org/rcsbsearch/v2/metadata/schema>`_.
-    Operators are documented `here <http://search.rcsb.org/#field-queries>`_.
-
-    The :py:class:`Attr` class provides a more pythonic way of constructing Terminals.
+    """A terminal query node for doing searches other than attribute searches.
+    Main function to allow for sequence searches. Does not have immediate use
     """
-
-    attribute: Optional[str] = None
-    service: str = STRUCTURE_ATTRIBUTE_SEARCH_SERVICE
-    operator: Optional[str] = None
-    value: Optional[TValue] = None
-    negation: Optional[bool] = False  # investigate whether this can be changed to None
+    service: str
+    params: Dict[str, Any]
     node_id: int = 0
 
     def to_dict(self):
-        params = dict()
-        if self.attribute is not None:
-            params["attribute"] = self.attribute
-        if self.operator is not None:
-            params["operator"] = self.operator
-        if self.value is not None:
-            params["value"] = self.value
-        if self.negation is not None:
-            params["negation"] = self.negation
-
         return dict(
             type="terminal",
             service=self.service,
-            parameters=params,
+            parameters=self.params,
             node_id=self.node_id,
         )
 
     def __invert__(self):
-        return Terminal(
-            self.attribute,
-            self.service,
-            self.operator,
-            self.value,
-            not self.negation,
-            self.node_id,
-        )
+        if isinstance(self, AttributeQuery):
+            return Terminal(self.service, {"attribute": self.params.get("attribute"),
+                                           "operator": self.params.get("operator"),
+                                           "negation": not self.params.get("negation"),
+                                           "value": self.params.get("value")})
+        else:
+            return Terminal(self.service, self.params)
 
     def _assign_ids(self, node_id=0) -> Tuple[Query, int]:
```

Comment:
No need to make this change now, but just wondering—do you think this method would better fit within `Session` or `Group`? So that it has access to all nodes that it could iterate over?